### PR TITLE
feat: Add minimum_cache_output_volume_bytes config setting

### PR DESCRIPTION
**Pull request by blaknite**
## Description

Adds a `minimum_cache_output_volume_bytes` runtime config setting that lets operators set a per-run capacity floor for cache output volumes on both the `firecracker` and `darwin-vz` backends. Previously the floor was a package-level constant with no way to override it without recompiling.

The resolver pattern follows `MinimumRootFSBytes`: the cfg field wins when non-zero, otherwise the package-level default applies. This preserves the existing test contract where both are zero-valued.

## Changes

- Add `MinimumCacheOutputVolumeBytes` field to `backend.FirecrackerConfig` and both runtimeconfig backend structs (`FirecrackerConfig`, `DarwinVZConfig`) under YAML key `minimum_cache_output_volume_bytes`
- Introduce `resolveCacheOutputVolumeMinimumBytes` (firecracker) and `resolveDarwinVZCacheOutputVolumeMinimumBytes` (darwin-vz) resolver helpers so cfg can override the package-level var without mutating it
- Wire the field through `MergeBackendConfig` — previously it was parsed but never copied into the output struct
- Introduce `prepareDarwinVZCacheOutputWritableVolumeFn` injection point on darwin-vz to make the writable-volume step testable, matching the existing `rootFSVolumeStoreDriverFn` pattern on firecracker
- Add tests covering integer bytes, human string (`16GiB`), invalid input rejection, explicit-zero override, and both call sites in the darwin-vz resolver
- Document the new setting in `docs/caching.md`

## Verification

Set `minimum_cache_output_volume_bytes: 16GiB` under `backends.darwin-vz` or `backends.firecracker` in the runtime config and confirm the cache output volume is created with at least that capacity.

## Rollback

Safe to revert — the new field is additive. Removing it restores the previous behaviour where the package-level default is always used.

### CODE REVIEW

File: internal/runtimeconfig/config.go [hunk 1]
```diff
@@ -369,6 +374,8 @@ func parseConfig(path string, raw []byte) (Config, error) {
 
 	darwinVZMinRootFSBytes := cfg.Backends.DarwinVZ.MinimumRootFSBytes
 	darwinVZMinRootFSBytesSet := presenceCfg.Backends.DarwinVZ.MinimumRootFSBytes != nil
+	darwinVZMinCacheOutputVolumeBytes := cfg.Backends.DarwinVZ.MinimumCacheOutputVolumeBytes
+	darwinVZMinCacheOutputVolumeBytesSet := presenceCfg.Backends.DarwinVZ.MinimumCacheOutputVolumeBytes != nil
```
[the remark this answers is not in the document]
Comment (blaknite) [reply internal/runtimeconfig/config.go:378]: Good catch — fixed in 6b9ee67. Extended `darwinVZConfigHasValues` to also consider `MinimumCacheOutputVolumeBytes`, matching the existing carve-out for `MinimumRootFSBytes`, and added a test that loads a legacy-key-only config with just this field set to prove the value survives.

**Comment by blaknite** [comment]
@codex please review again
